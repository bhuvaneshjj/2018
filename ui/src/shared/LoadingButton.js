import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import LoadingIndicator from './LoadingIndicator';

const LoadingButton = ({ bsStyle = 'success', type, loading = false, text, loadingText }) => (
  <Button type={type} bsStyle={bsStyle} disabled={loading}>
    {loading ? (
      <span>
        <LoadingIndicator /> {loadingText}
      </span>
    ) : (
      text
    )}
  </Button>
);

LoadingButton.propTypes = {
  bsStyle: PropTypes.string,
  type: PropTypes.string,
  loading: PropTypes.bool,
  text: PropTypes.string.isRequired,
  loadingText: PropTypes.string
};

export default LoadingButton;
