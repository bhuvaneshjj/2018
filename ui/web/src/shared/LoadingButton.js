import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import LoadingIndicator from './LoadingIndicator';

const LoadingButton = ({
  bsStyle = 'success',
  children,
  type,
  loading = false,
  loadingText,
  onClick
}) => (
  <Button type={type} bsStyle={bsStyle} disabled={loading} onClick={onClick}>
    {loading ? (
      <span>
        <LoadingIndicator /> {loadingText}
      </span>
    ) : (
      children
    )}
  </Button>
);

LoadingButton.propTypes = {
  bsStyle: PropTypes.string,
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
  loading: PropTypes.bool,
  loadingText: PropTypes.string,
  onClick: PropTypes.func
};

export default LoadingButton;
