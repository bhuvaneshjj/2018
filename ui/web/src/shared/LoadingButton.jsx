import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import LoadingIndicator from './LoadingIndicator';

const LoadingButton = ({ bsStyle, children, type, loading, loadingText, onClick }) => (
  <Button type={type} bsStyle={bsStyle} disabled={loading} onClick={onClick}>
    {loading ? (
      <span>
        <LoadingIndicator /> {loadingText}
      </span>
    ) : (
      children
    )}
  </Button>
);

LoadingButton.propTypes = {
  bsStyle: PropTypes.string,
  children: PropTypes.node.isRequired,
  type: PropTypes.string,
  loading: PropTypes.bool,
  loadingText: PropTypes.string, // eslint-disable-line react/require-default-props
  onClick: PropTypes.func // eslint-disable-line react/require-default-props
};

LoadingButton.defaultProps = {
  bsStyle: 'success',
  loading: false,
  type: 'button'
};

export default LoadingButton;
