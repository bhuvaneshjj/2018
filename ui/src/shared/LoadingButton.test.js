import React from 'react';
import renderer from 'react-test-renderer';
import LoadingButton from './LoadingButton';

it('renders - not loading', () => {
  const component = renderer.create(<LoadingButton text="čudl" />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('renders - loading', () => {
  const component = renderer.create(<LoadingButton loading={true} text="čudl" />);
  expect(component.toJSON()).toMatchSnapshot();
});
