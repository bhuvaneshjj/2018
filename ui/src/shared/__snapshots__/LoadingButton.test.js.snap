// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders - loading 1`] = `
<button
  className="btn btn-success"
  disabled={true}
  type="button"
>
  <span>
    <span
      className="LoadingIndicator glyphicon glyphicon-repeat"
    />
     
  </span>
</button>
`;

exports[`renders - not loading 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  type="button"
>
  čudl
</button>
`;
